### PR TITLE
collect dependency for conceptual document.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @superyyrrzz @hellosnow @ansyral @qinezh 
#449

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/DependencyGraph.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+namespace Microsoft.DocAsCode.Build.Engine
+{
+    using System.Collections.Generic;
+
+    public class DependencyGraph
+    {
+        private readonly Dictionary<string, SortedSet<string>> _dictionary =
+            new Dictionary<string, SortedSet<string>>();
+
+        public void ReportDependency(string key, string value)
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/DependencyGraph.cs]: reuse the overload method below?

**Comment by hellosnow** [comment]
:shipit:

**Comment by chenkennt** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/DependencyGraph.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                    if (_dictionary.TryGetValue(current, out set))
+                    {
+                        foreach (var item in set)
+                        {
+                            queue.Enqueue(item);
+                        }
+                    }
+                }
+            }
+            return result;
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/DependencyGraph.cs]: Can we record the result for later use to avoid some repeated computation?

Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/DependencyGraph.cs]: I want to keep it simple, if you think cache is necessary, I will create a cache decorator for it.

**Comment by superyyrrzz** [comment]
:+1:

**Pull request merged**